File: src/Python/easy/string/problem796.py


class Solution:
    def rotateString(self, s: str, goal: str) -> bool:
        res = None
        for i in range(len(s)):
            s = "".join(s[1:len(s)] + s[0:1]) 
            if s == goal:
                res = True
                break
            else:
                res = False
        return res
    
    
s = "abcde"; goal = "cdeab"

solution = Solution()
solution.rotateString(s, goal)

File: src/JavaScript/easy/string/problem796.js
/**
 * @param {string} s
 * @param {string} goal
 * @return {boolean}
 */
var rotateString = function (s, goal) {
  let res = null;
  for (let i = 0; i < s.length; i++) {
    s = [...s.split("").splice(1, s.length), ...s.split("").splice(0, 1)].join("");
    if (s == goal) {
      res = true;
      break;
    } else {
      res = false;
    }
  }
  return res
};

// let s = "abcde", goal = "cdeab";
let s = "abcde", goal = "abced"

rotateString(s, goal);
